mcs/class/corlib/System/MulticastDelegate.cs

```diff
@@ -61,13 +60,20 @@ public override void GetObjectData (SerializationInfo info, StreamingContext con
 			base.GetObjectData  (info, context);
 		}
 
-
 		protected sealed override object DynamicInvokeImpl (object[] args)
 		{
-			if (prev != null)
-				prev.DynamicInvokeImpl (args);
-
-			return base.DynamicInvokeImpl (args);
+			if (delegates == null) {
+				return base.DynamicInvokeImpl (args);
+			} else {
+				int len = delegates.Length;
+				if (len >= 1) {
+					object r = null;
+					for (int i = 0; i < len; ++i)
+						r = delegates [i].DynamicInvoke (args);
+					return r;
```

Comment:
Yes, multicast return value behavior is weird not not useful at all.

One wonder why allow it for delegates that return value.